modules/ROOT/pages/dataweave-selecting-types.adoc

```diff
@@ -0,0 +1,204 @@
+= Selecting Types
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Using the dot selector (`.`) over DataWeave xref:dataweave-type-system.adoc#composite-types[types] enables you to declare new types from existing ones. Combined with the xref:dataweave-selectors-reusing-types.adoc[module loader], DataWeave can also load and translate declarations from a custom module file into DataWeave type directives that can be accessed in the same way as types from any other DataWeave module.
+
+== Declare a New Type from an Existing Type
+The following example declares the `NameType` from the `User` type. Using the dot selector in `User.name` to declare the `name` variable a `NameType` enables the `typeOf()`  function to return the basic type, `"String"`. Notice how the DataWeave variable `name` is declared as a `NameType` value "Seba":
```

Comment:
```suggestion
The following example declares the `NameType` from the `User` type. Using the dot selector in `User.name` to declare the `name` variable a `NameType` enables the `typeOf()`  function to return the primitive type, `"String"`. Notice how the DataWeave variable `name` is declared as a `NameType` value "Seba":
```

Reply:
changed to "primitive" as mentioned by dev and since it's universally understood